services/port_manager/src/main/java/com/futurewei/alcor/portmanager/controller/PortController.java

```diff
@@ -203,4 +205,22 @@ public PortWebBulkJson listPort(@PathVariable("project_id") String projectId) th
 
         return new PortWebBulkJson(portsList);
     }
+
+    /**
+     * Update neighbor tables and send them to DPM when adding or deleting gateway port
+     * @param projectId Project Id
+     * @param routerSubnetUpdateInfo Router's latest subnet information
+     * @return RouterSubnetUpdateInfo
+     * @throws Exception Db operation exception
+     */
+    @Rbac(resource ="port")
+    @PostMapping({"/project/{project_id}/update-l3-neighbors", "v4/{project_id}/update-l3-neighbors"})
+    @ResponseBody
+    @ResponseStatus(HttpStatus.CREATED)
+    @DurationStatistics
+    public RouterSubnetUpdateInfo updateL3Neighbors(@PathVariable("project_id") String projectId,
+                                                    @RequestBody RouterSubnetUpdateInfo routerSubnetUpdateInfo) throws Exception {
+        checkRouterSubnetUpdateInfo(routerSubnetUpdateInfo);
+        return portService.updateL3Neighbors(projectId, routerSubnetUpdateInfo);
+    }
 }
```

Comment:
I suggest that using PUT for this api, then it will become /project/{project_id}/update-l3-neighbors/{new_subnet_id}.
The BODY becomes {operation_type, vpcid, [old_subnet_ids]}

Reply:
ok